src/ui/run-history-ui-handler.ts

```diff
@@ -0,0 +1,317 @@
+  private loadingLabel: Phaser.GameObjects.Text;
+
+  constructor(scene: BattleScene, entryData: RunEntry, slotId: number) {
+    super(scene, 0, slotId*56);
+
+    this.slotId = slotId;
+    this.hasData = true;
+    this.entryData = entryData;
+
+    this.setup(this.entryData);
+
+  }
+
+  setup(run: RunEntry) {
+
+    const victory = run.isVictory;
+    const data = this.scene.gameData.parseSessionData(JSON.stringify(run.entry));
+
+    const slotWindow = addWindow(this.scene, 0, 0, 304, 52);
+    this.add(slotWindow);
+
+
+    if (victory) {
+      const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:victory")}`, TextStyle.WINDOW);
+      this.add(gameOutcomeLabel);
+    } else {
+      if (data.battleType === BattleType.WILD) {
+        const enemyContainer = this.scene.add.container(8, 5);
+        const gameOutcomeLabel = addTextObject(this.scene, 0, 0, `${i18next.t("runHistory:defeatedWild")}`, TextStyle.WINDOW);
+        enemyContainer.add(gameOutcomeLabel);
+        data.enemyParty.forEach((enemyData, e) => {
+          // This allows the enemyParty to be shown - doubles or singles -> 58+(e*8)
+          const enemyIconContainer = this.scene.add.container(65+(e*25),-8);
+          enemyIconContainer.setScale(0.75);
+          enemyData.boss = false;
+          enemyData["player"] = true;
+          const enemy = enemyData.toPokemon(this.scene);
+          const enemyIcon = this.scene.addPokemonIcon(enemy, 0, 0, 0, 0);
+          const enemyLevel = addTextObject(this.scene, 32, 20, `${i18next.t("saveSlotSelectUiHandler:lv")}${Utils.formatLargeNumber(enemy.level, 1000)}`, TextStyle.PARTY, { fontSize: "54px", color: "#f8f8f8" });
+          enemyLevel.setShadow(0, 0, undefined);
+          enemyLevel.setStroke("#424242", 14);
+          enemyLevel.setOrigin(1, 0);
+          enemyIconContainer.add(enemyIcon);
+          enemyIconContainer.add(enemyLevel);
+          enemyContainer.add(enemyIconContainer);
+          enemy.destroy();
+        });
+        this.add(enemyContainer);
+      } else if (data.battleType === BattleType.TRAINER) {
+        const tObj = data.trainer.toTrainer(this.scene);
+        const RIVAL_TRAINER_ID_THRESHOLD = 375;
+        if (data.trainer.trainerType >= RIVAL_TRAINER_ID_THRESHOLD) {
+          const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:defeatedRival")}`, TextStyle.WINDOW);
+          // otherwise it becomes Rival_5 in Ivy's case
+          this.add(gameOutcomeLabel);
+        } else if (data.trainer.variant === TrainerVariant.DOUBLE) {
+          const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:defeatedTrainer")+tObj.config.nameDouble+" "+tObj.getName(0, false)}`, TextStyle.WINDOW);
+          this.add(gameOutcomeLabel);
+        } else {
+          const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:defeatedTrainer")+tObj.getName(0, true)}`, TextStyle.WINDOW);
```

Comment:
NIT: odd use of template string.
Either
```suggestion
          const gameOutcomeLabel = addTextObject(this.scene, 8, 5, `${i18next.t("runHistory:defeatedTrainer")}${tObj.getName(0, true)}`, TextStyle.WINDOW);
```
or
```suggestion
          const gameOutcomeLabel = addTextObject(this.scene, 8, 5, i18next.t("runHistory:defeatedTrainer")  + tObj.getName(0, true), TextStyle.WINDOW);
```